File: android/src/main/java/com/margelo/nitro/googlecast/converters/GckMediaSeekOptions+toMediaSeekOptions.kt
package com.margelo.nitro.googlecast.converters

import com.margelo.nitro.googlecast.MediaSeekOptions
import com.margelo.nitro.googlecast.MediaSeekResumeState
import com.google.android.gms.cast.MediaSeekOptions as GckMediaSeekOptions

/**
 * Converts a Google Cast [GckMediaSeekOptions] into a generated [MediaSeekOptions] struct.
 *
 * Reverse of `MediaSeekOptions+toGckMediaSeekOptions.kt`. GCK milliseconds become seconds.
 * `resumeState` is mapped BY VALUE; `RESUME_STATE_UNCHANGED` becomes `null`. `relative` has no
 * GCK source and is always `null`.
 */
internal fun GckMediaSeekOptions.toMediaSeekOptions(): MediaSeekOptions {
  val mappedResume = when (resumeState) {
    GckMediaSeekOptions.RESUME_STATE_PLAY -> MediaSeekResumeState.PLAY
    GckMediaSeekOptions.RESUME_STATE_PAUSE -> MediaSeekResumeState.PAUSE
    else -> null
  }
  return MediaSeekOptions(
    position = position / 1000.0,
    relative = null,
    infinite = isSeekToInfinite,
    resumeState = mappedResume,
    customData = customData?.toAnyMap()
  )
}

File: ios/NitroGoogleCast/Converters/GCKMediaSeekOptions+toMediaSeekOptions.swift
import Foundation
import GoogleCast
import NitroModules

/// Converts a `GCKMediaSeekOptions` into a generated `MediaSeekOptions` struct.
///
/// Reverse of `MediaSeekOptions+toGckMediaSeekOptions.swift`. GCK's `interval`/`relative`/
/// `seekToInfinite` are non-optional, so `position`/`relative`/`infinite` are always emitted
/// (the corpus pins GCK's defaults for the absent case). `resumeState` `Unchanged` → `nil`.
extension GCKMediaSeekOptions {
  func toMediaSeekOptions() -> MediaSeekOptions {
    return MediaSeekOptions(
      position: interval,
      relative: relative,
      infinite: seekToInfinite,
      resumeState: resumeState.toMediaSeekResumeState(),
      customData: AnyMap.fromGckCustomData(customData)
    )
  }
}
